internal/codeintel/reposcheduler/store.go

```diff
@@ -0,0 +1,289 @@
+package reposcheduler
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/keegancsmith/sqlf"
+	logger "github.com/sourcegraph/log"
+	"github.com/sourcegraph/sourcegraph/internal/database"
+	"github.com/sourcegraph/sourcegraph/internal/database/basestore"
+	"github.com/sourcegraph/sourcegraph/internal/database/dbutil"
+	"github.com/sourcegraph/sourcegraph/internal/observation"
+	"go.opentelemetry.io/otel/attribute"
+)
+
+type RepoRev struct {
+	ID           int
+	RepositoryID int
+	Rev          string
+}
+
+type RepositorySchedulingStore interface {
+	WithTransaction(ctx context.Context, f func(tx RepositorySchedulingStore) error) error
+	GetRepositoriesForIndexScan(ctx context.Context, batchOptions RepositoryBatchOptions, now time.Time) ([]RepositoryToIndex, error)
+}
+
+type store struct {
+	db         *basestore.Store
+	logger     logger.Logger
+	operations *operations
+	dbLayout   dbLayout
+}
+
+var _ RepositorySchedulingStore = &store{}
+
+type StoreType int8
+
+const (
+	Precise   StoreType = 1
+	Syntactic StoreType = 2
+)
```

Comment:
```suggestion
type storeType int8

const (
	preciseStore   StoreType = 1
	syntacticStore StoreType = 2
)
```

Suggestion: Make these private, since AFAICT, the values are only used within this same package.